lib/sdk/storage.js

```diff
@@ -47,26 +46,40 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         }
         console.log("WORKER_TEST_MODE: asset path:", filePath);
         return new Asset(assetReference, directory, name);
+    }
 
-    } else {
-        const source = new Asset(assetReference, directory, name);
+    const source = new Asset(assetReference, directory, name);
+    if (isDataUrlAndValid(assetReference.url)) {
```

Comment:
is this code copied from the SDK?
please add a note if its just "syncing with SDK code"
or if there are additional changes, add a note so we know about them when we combine these into one central place

Reply:
done